sdk/core/Azure.Core.Amqp/src/AmqpMessageBody.cs

```diff
@@ -16,7 +16,17 @@ public class AmqpMessageBody
         /// <summary>
         /// The data sections for the AMQP message body.
         /// </summary>
-        private readonly IEnumerable<ReadOnlyMemory<byte>> _data;
+        private readonly IEnumerable<ReadOnlyMemory<byte>>? _data;
+
+        /// <summary>
+        /// The value section for the AMQP message body.
+        /// </summary>
+        private readonly object? _value;
+
+        /// <summary>
+        /// The sequence sections for the AMQP message body.
+        /// </summary>
+        private IEnumerable<IList<object>>? _sequence;
```

Comment:
This is an interesting composition.   I'm curious why we decided on an enumerable of lists rather than `IList<IList<object>>` or `IEnumerable<IEnumerable<object>>`.

Reply:
It seemed overly complicated to have an IEnumerable<IEnumerable> as in that case you are not only generating the outer lists but the inner lists as well. IList<IList> felt too restrictive and doesn't follow our general guidance around accepting the the most general type as input.